test/JDBC/expected/1_grant_schema.out

```diff
@@ -0,0 +1,2936 @@
+-- tsql
+-- Test group 1 [guest users and dbo schema]
+-- create objects
+create login l1 with password = '12345678'
+go
+create table t1(a int);
+go
+create view v1 as select 2;
+go
+create proc p1 as select 1;
+go
+create function f1() returns int begin declare @a int; set @a = 1; return @a; end 
+go
+
+-- tsql user=l1 password=12345678
+-- guest user should have no permission
+select current_user;
+go
+~~START~~
+varchar
+guest
+~~END~~
+
+select * from dbo.t1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+select * from dbo.v1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for view v1)~~
+
+exec dbo.p1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for procedure p1)~~
+
+select * from dbo.f1()
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for function f1)~~
+
+
+-- tsql
+grant select, execute on schema::dbo to guest;
+go
+create function f2() returns int begin declare @a int; set @a = 1; return @a; end 
+go
+
+-- tsql user=l1 password=12345678
+-- guest user should have select and EXECUTE permission
+select current_user;
+go
+~~START~~
+varchar
+guest
+~~END~~
+
+select * from dbo.t1;
+go
+~~START~~
+int
+~~END~~
+
+select * from dbo.v1;
+go
+~~START~~
+int
+2
+~~END~~
+
+exec dbo.p1;
+go
+~~START~~
+int
+1
+~~END~~
+
+select * from dbo.f1()
+go
+~~START~~
+int
+1
+~~END~~
+
+select * from dbo.f2()
+go
+~~START~~
+int
+1
+~~END~~
+
+
+-- tsql
+grant select on dbo.t1 to guest;
+go
+grant execute on dbo.f1 to guest;
+go
+
+-- tsql user=l1 password=12345678
+-- guest user should have select and EXECUTE permission
+select current_user;
+go
+~~START~~
+varchar
+guest
+~~END~~
+
+select * from dbo.t1;
+go
+~~START~~
+int
+~~END~~
+
+select * from dbo.v1;
+go
+~~START~~
+int
+2
+~~END~~
+
+exec dbo.p1;
+go
+~~START~~
+int
+1
+~~END~~
+
+select * from dbo.f1()
+go
+~~START~~
+int
+1
+~~END~~
+
+select * from dbo.f2()
+go
+~~START~~
+int
+1
+~~END~~
+
+
+-- tsql
+revoke select on dbo.t1 from guest;
+go
+revoke execute on dbo.f1 from guest;
+go
+revoke select, execute on schema::dbo to guest;
```

Comment:
does it make sense to check if we still have access objects after object level revoke? If we already have tests like for this then we can ignore this

Reply:
We already have tests like this in the sections below.